notebook-lessons/oss-module-08-foss-code-life.ipynb

```diff
@@ -0,0 +1,391 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e9b27e8c",
+   "metadata": {},
+   "source": [
+    "# Open Source Lifecycle and Sustainability 🌍📦\n",
+    "\n",
+    "# 🧠 Overview\n",
+    "This notebook walks through the natural lifecycle of an Open Source Software (OSS) project and provides practical guidance, tools, and an interactive utility to evaluate OSS project sustainability using your GitHub's metadata.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41eb5afe",
+   "metadata": {},
+   "source": [
+    "## 🔄 1. Natural Open Source Lifecycle\n",
+    "### 📌 Open Source Project Lifecycle: Project → Ecosystem → Archive\n",
+    "\n",
+    "#### 🚀 Project Phase\n",
+    "- Initial development, usually by one or a few individuals.\n",
+    "- Focused scope, solving a specific problem.\n",
+    "- Examples: first commit, README, LICENSE.\n",
+    "\n",
+    "#### 🌐 Ecosystem Phase\n",
+    "- Community forms, contributions increase.\n",
+    "- Features grow, integrations appear.\n",
+    "- Tools like GitHub Issues, Discussions, Actions, and Projects are used.\n",
+    "\n",
+    "#### 🗂️ Archive Phase\n",
+    "- Maintainers shift priorities, activity drops.\n",
+    "- Project is deprecated, forked, or archived.\n",
+    "- A good exit strategy includes archive notices and project transfer guidance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8bb2038b",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## 🌱 2. Sustainability Practices for Each Lifecycle Phase\n",
+    "\n",
+    "### 🚀 Project Phase\n",
+    "- **Actionable Steps**:\n",
+    "  - Define a clear project scope and goals.\n",
+    "  - Create essential files: `README.md`, `LICENSE`, and `CODE_OF_CONDUCT.md`.\n",
+    "  - Use GitHub Issues to track bugs and features.\n",
+    "\n",
+    "### 🌐 Ecosystem Phase\n",
+    "- **Actionable Steps**:\n",
+    "  - Foster community engagement through GitHub Discussions.\n",
+    "  - Automate workflows with GitHub Actions.\n",
+    "  - Provide detailed documentation using tools like MkDocs or ReadTheDocs.\n",
+    "\n",
+    "### 🗂️ Archive Phase\n",
+    "- **Actionable Steps**:\n",
+    "  - Announce deprecation with clear archive notices.\n",
+    "  - Provide guidance for forks and project transfers.\n",
+    "  - Maintain transparency with contributors and users.\n",
+    "\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "359081f7",
+   "metadata": {},
+   "source": [
+    "## 🛠️ 3. Techniques and Tools to Increase OSS Sustainability\n",
+    "\n",
+    "### ✅ Key Sustainability Practices:\n",
+    "- ✅ Clear README, LICENSE, and CODE_OF_CONDUCT.md\n",
+    "- ✅ CONTRIBUTING.md to guide newcomers\n",
+    "- ✅ GitHub Actions for CI/CD\n",
+    "- ✅ Good Issue and PR templates\n",
+    "- ✅ Documentation hosted with tools like MkDocs or ReadTheDocs\n",
```

Comment:
MkDocs doesn't provide hosting capabilities, as far as I know.  Distinguish between documentation generation (mkdocs, sphinx, etc) and documentation hosting (readthedocs, GitHub pages)